src/sidebar/store/modules/session.js

```diff
@@ -1,38 +1,40 @@
 import * as util from '../util';
 
 function init() {
-  /**
-   * Profile/session information for the active user.
-   */
   return {
-    /** A map of features that are enabled for the current user. */
-    features: {},
-    /** A map of preference names and values. */
-    preferences: {},
     /**
-     * The authenticated user ID or null if the user is not logged in.
+     * Profile object fetched from the `/api/profile` endpoint.
      */
-    userid: null,
+    profile: {
```

Comment:
This object doesn't yet initialize all of the properties that end up on it, e.g. `user_info`, which is accessed by the `hypothesis-app` and `AnnotationUser` components, e.g.

This brings up a higher-level thing to think about, to do with typedefs and how we codify what we "expect" objects provided by an external service to look like...

Reply:
> This brings up a higher-level thing to think about, to do with typedefs and how we codify what we "expect" objects provided by an external service to look like...

I have some ideas on that front. This would be a good topic to discuss in a future frontend sync or part of our general future technical planning once we've finished our current a11y and Preact migration projects.